sdk/core/src/main/java/com/klaviyo/coresdk/model/Profile.kt

```diff
@@ -54,39 +46,60 @@ class Profile(properties: Map<ProfileKey, Serializable>?) :
         this[ProfileKey.CUSTOM(key)] = value
     }
 
-    override fun merge(other: Profile) = apply {
-        super.merge(other).also {
-            other.appendMap.forEach { (k, v) -> addAppendProperty(k, v) }
-        }
-    }
+    override fun merge(other: Profile) = apply { super.merge(other) }
+
+    /**
+     * Get a map of just the unique identifiers of this profile
+     */
+    internal fun getIdentifiers(): Map<ProfileKey, String> = mapOf(
+        ProfileKey.EXTERNAL_ID to (this.externalId ?: ""),
+        ProfileKey.EMAIL to (this.email ?: ""),
+        ProfileKey.PHONE_NUMBER to (this.phoneNumber ?: ""),
+        ProfileKey.ANONYMOUS_ID to (this.anonymousId ?: ""),
+    ).filter { it.value.isNotEmpty() }
 }
 
 /**
- * All profile property keys recognised by the Klaviyo APIs
+ * All profile attributes recognised by the Klaviyo APIs
  * Custom properties can be defined using the [CUSTOM] inner class
  */
 sealed class ProfileKey(name: String) : Keyword(name) {
 
     // Identifiers
-    object EXTERNAL_ID : ProfileKey("\$external_id")
-    object EMAIL : ProfileKey("\$email")
-    object PHONE_NUMBER : ProfileKey("\$phone_number")
-    internal object ANONYMOUS : ProfileKey("\$anonymous")
+    object EXTERNAL_ID : ProfileKey("external_id")
+    object EMAIL : ProfileKey("email")
+    object PHONE_NUMBER : ProfileKey("phone_number")
+    internal object ANONYMOUS_ID : ProfileKey("anonymous_id")
 
     // Personal information
-    object FIRST_NAME : ProfileKey("\$first_name")
-    object LAST_NAME : ProfileKey("\$last_name")
-    object TITLE : ProfileKey("\$title")
-    object ORGANIZATION : ProfileKey("\$organization")
-    object CITY : ProfileKey("\$city")
-    object REGION : ProfileKey("\$region")
-    object COUNTRY : ProfileKey("\$country")
-    object ZIP : ProfileKey("\$zip")
-    object IMAGE : ProfileKey("\$image")
+    object FIRST_NAME : ProfileKey("first_name")
+    object LAST_NAME : ProfileKey("last_name")
+    object ORGANIZATION : ProfileKey("organization")
+    object TITLE : ProfileKey("title")
+    object IMAGE : ProfileKey("image")
+
+    object ADDRESS1 : ProfileKey("address1")
+    object ADDRESS2 : ProfileKey("address2")
+    object CITY : ProfileKey("city")
+    object COUNTRY : ProfileKey("country")
+    object LATITUDE : ProfileKey("latitude")
+    object LONGITUDE : ProfileKey("longitude")
+    object REGION : ProfileKey("region")
+    object ZIP : ProfileKey("zip")
+    object TIMEZONE : ProfileKey("timezone")
 
     // Custom properties
     class CUSTOM(propertyName: String) : ProfileKey(propertyName)
 
-    // Other
-    internal object APPEND : ProfileKey("\$append")
+    /**
+     * Helper method to translate certain keys to their dollar-prefixed key
+     * This only applies to the identifier keys under certain circumstances
+     *
+     * @return
+     */
+    internal fun specialKey(): String = when (this) {
```

Comment:
Sorry what's this for? Is it because events API still uses the old $ keys but V3 profiles API does not? So you need to do some simple conversion?

Reply:
Exactly right: profile V2 and events V3 and use `$`, profile V3 does not use `$`. I need the conversion for these identifier fields so that the event and push token API calls can get associated with the correct profile. 

I'm going to copy that comment into this spot in the repo